### PR TITLE
Friends interface rework

### PR DESCRIPTION
**Pull request by caillef**

### CODE REVIEW

File: lua/modules/uikit.lua [hunk 1]
```diff
@@ -660,6 +663,39 @@ function createUI(system)
...
+		elseif k == "onClick" then
+			if t.type == NodeType.Button then
+				t._onClick = function(self)
+					_buttonOnRelease(self, v)
+				end
+			elseif t.type == NodeType.Frame then
+				local background = t.background
+				if v == nil then
+					t._onClick = nil
+					if t._onPress == nil then
```
Comment (aduermael) [inline lua/modules/uikit.lua:675]: what if _onRelease is set?
We should not disable collisions

Comment (aduermael) [reply lua/modules/uikit.lua:675]: _onClick should also be tested when setting _onRelease and _onPress to nil

File: lua/modules/uikit.lua [hunk 2]
```diff
@@ -2864,6 +2957,7 @@ function createUI(system)
 		end
 
 		local pressed = pressed
+		pressed._dragging = true
```
Comment (aduermael) [inline lua/modules/uikit.lua:2960]: Should move in the block below, `pressed` could be `nil`

Comment (aduermael) [reply lua/modules/uikit.lua:2960]: Also, there's a `CLICK_MOVE_SQR_EPSILON` in `controls`. Same logic should be used here. 
On mobile, very small, unwanted drag events can happen a lot (depending on devices)

**Review by aduermael — CHANGES_REQUESTED** [review]

File: lua/modules/uikit.lua (hunk 2 above)
Comment (caillef) [reply lua/modules/uikit.lua:2960]: I think the best solution is to change how the PointerDrag is triggered.
Today, controls listen to PointerDrag and calls Pointer.DragBegin/Pointer.Drag/Pointer.DragEnd
which means that listening on PointerDrag or adding a Pointer.Drag callback is not the same behavior.

I think the best solution is to have a SystemPointerDrag listened by controls, and this calls Pointer.Drag + send PointerDrag events for games to avoid this issue, as I think this should not be handled individually.

Let me know if I should create an issue, not sure it goes in this PR.

File: lua/modules/uikit.lua (hunk 1 above)
Comment (caillef) [reply lua/modules/uikit.lua]: I just updated this, thanks for finding that one!

**Review by aduermael — COMMENTED** [review]
Found a few bugs: 

- Clicking outside the window activates cells, we should probably check collision agains scroll area before propagating to content
- There's one invisible row at the end:
    Screenshot 2024-01-22 at 16 16 32
- I can't see sent requests (that section should be folded by default)

**Comment by aduermael** [comment]
- **✅ FIXED** Clicking outside the window activates cells, we should probably check collision agains scroll area before propagating to content
- **✅ FIXED** Can't drag scroll with mouse drag
- **✅ FIXED** drag should continue even outside of frame as long as pointer down happened within

**Review by aduermael — APPROVED** [review]
I think I've fixed the most important issues. 
I could even remove `onClick` from `uikit` (let's discuss this)
